src/adapter/src/coord.rs

```diff
@@ -1611,6 +1617,20 @@ pub struct Coordinator {
     /// it manually.
     advance_timelines_interval: tokio::time::Interval,
 
+    /// Generic deferred statements.
```

Comment:
It's a bit confusing that this is described as generic for all statements, but we also have 
```Rust
/// Holds plans deferred due to write lock.
write_lock_wait_group: VecDeque<Deferred>,
```
above, which are for deferred inserts, which is a type of statement. Maybe we can add a note that inserts are excluded because they predate deferred statements. Also there seems to be some overlap in `DeferredPlan`

```Rust
/// This is the struct meant to be paired with [`Message::WriteLockGrant`], but
/// could theoretically be used to queue any deferred plan.
#[derive(Derivative)]
#[derivative(Debug)]
pub(crate) struct DeferredPlan {
    #[derivative(Debug = "ignore")]
    pub ctx: ExecuteContext,
    pub plan: Plan,
    pub validity: PlanValidity,
}
```

and `DeferredStatement` which both claim to be generic over any statement/plan type.